src/components/Switch/Switch.spec.tsx

```diff
@@ -0,0 +1,38 @@
+import { render, screen } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+
+import Switch from './Switch';
+
+describe('Switch', () => {
+  describe('when initialize', () => {
+    it('should render the switch off', () => {
+      render(<Switch checked={false} setChecked={() => {}} />);
+
+      const switchComponent = screen.getByRole('checkbox');
+
+      expect(switchComponent).not.toBeChecked();
+      expect(switchComponent).toBeInTheDocument();
+    });
+
+    it('should render the switch off', () => {
+      render(<Switch checked={true} setChecked={() => {}} />);
+
+      const switchComponent = screen.getByRole('checkbox');
+
+      expect(switchComponent).toBeChecked();
+      expect(switchComponent).toBeInTheDocument();
+    });
+  });
+
+  describe('when click', () => {
+    it('should call the setChecked function ', () => {
+      const setCheckedMock = jest.fn();
+      render(<Switch checked={true} setChecked={setCheckedMock} />);
+
+      const switchComponent = screen.getByRole('checkbox');
+      userEvent.click(switchComponent);
+
+      expect(setCheckedMock).toHaveBeenCalledTimes(1);
```

Comment:
Isso não ta disparando nenhum warning? Geralm ente quando usamos eventos temos que usar a função `act()` com `await`. Da uma olhada, se for realmente desnecessária tudo bem, só to curioso

Reply:
pior que não ta dando nenhum warning, eu achei estranho não precisar transformar o teste assíncrono mas ele ta passando sem nenhum warning, sobre o act, eu li um [artigo sobre o criador do RTL](https://kentcdodds.com/blog/common-mistakes-with-react-testing-library) dizendo que hoje, dentro do  `fireEvent` e do `render` ele já usa o `act` internamente então é redundante usar fora, como o `userEvent` é feito em cima do `fireEvent`, eu acredito que também não é necessário o `act` 

![image](https://github.com/Alecell/octopost/assets/103784814/ec514d9e-18fc-49af-a677-d1d37ad95668)

![image](https://github.com/Alecell/octopost/assets/103784814/6185c04b-5de3-4d83-a331-b663cad48a04)

Reply:
resposta de qualidade 🙏